### PR TITLE
Don't skip messages, which exceeds gas allowance

### PR DESCRIPTION
**Pull request by breathx**
@gear-tech/dev

### CODE REVIEW

**Review by NikVolf — APPROVED** [review]

File: core-processor/src/processor.rs [hunk 1]
```diff
@@ -259,11 +271,41 @@ pub fn process_executable<A: ProcessorExt + EnvExt + IntoExtInfo + 'static, E: E
...
+    dispatch: IncomingDispatch,
+    program_id: ProgramId,
+    gas_burned: u64,
+) -> Vec<JournalNote> {
+    let mut journal = Vec::with_capacity(1);
+
+    let dispatch = StoredDispatch::new(
+        dispatch.kind(),
+        dispatch.message().clone().into_stored(program_id),
+        dispatch.context().clone(),
```
Comment (gshep) [inline core-processor/src/processor.rs:298]: do we really need to clone here?

**Review by gshep — APPROVED** [review]

File: core-processor/src/executor.rs [hunk 2]
```diff
@@ -77,11 +89,22 @@ pub fn execute_wasm<A: ProcessorExt + EnvExt + IntoExtInfo + 'static, E: Environ
...
+                allowance_exceed: true,
+            });
+        };
+
         if gas_counter.charge(amount) != ChargeResult::Enough {
             return Err(ExecutionError {
                 program_id,
                 gas_amount: gas_counter.into(),
                 reason: "Not enough gas for grow memory size.",
+                allowance_exceed: true,
```
[the remark this answers is not in the document]
Comment (breathx) [reply core-processor/src/executor.rs:107]: Fixed in #827

File: core-processor/src/processor.rs (hunk 1 above)
Comment (breathx) [reply core-processor/src/processor.rs:298]: Fixed in #827

File: core-processor/src/executor.rs [hunk 3]
```diff
@@ -77,11 +89,22 @@ pub fn execute_wasm<A: ProcessorExt + EnvExt + IntoExtInfo + 'static, E: Environ
...
             settings.mem_grow_cost() * (max_page as u64 + 1 - program.static_pages() as u64);
+
+        if gas_allowance_counter.charge(amount) != ChargeResult::Enough {
+            return Err(ExecutionError {
+                program_id,
+                gas_amount: gas_counter.into(),
+                reason: "",
+                allowance_exceed: true,
+            });
+        };
```
[the remark this answers is not in the document]
Comment (breathx) [reply core-processor/src/executor.rs:100]: Fixed in #827

**Pull request merged**